### PR TITLE
get rid of redundant id parameter for Model delete method

### PR DESCRIPTION
**Pull request by 66eli77**

### CODE REVIEW

**Comment by indirectlylit** [comment]
@ralphiee22 if you want to review this I'll commit it when you've approved

**Comment by jamalex** [comment]
This looks good, as long as it's covered all the cases of calling delete while passing an id. Did you search around for any others using that mode, @66eli77?

**Comment by 66eli77** [comment]
@jamalex nothing in the current branch uses the delete.

**Comment by indirectlylit** [comment]
```
/Users/d/Projects/learning_equality/repos/kolibri/kolibri/core/assets/src/core-actions.js:
   23    const id = 'current';
   24    const sessionModel = SessionResource.getModel(id);
   25:   const logoutPromise = sessionModel.delete(id);
   26    logoutPromise.then((response) => {
   27      store.dispatch('CORE_CLEAR_SESSION');
```

File: kolibri/core/assets/src/api-resource.js [hunk 1]
```diff
@@ -186,7 +186,7 @@ class Model {
             this.resource.removeModel(this);
```
Comment (rtibbles) [inline kolibri/core/assets/src/api-resource.js:184]: This section doesn't actually make sense.

If the model doesn't have an id, then it won't be in the model cache.

Also, then resolving it with `this.id` which we know is not defined inside this `if` block also makes no sense.

Might be better to raise an error here saying "Can't delete model that we don't have an id for".

**Comment by ralphiee22** [comment]
@66eli77 you probably want to pull in latest master and check again for cases of delete(id). I think some more cases of this have been added

File: kolibri/core/assets/src/api-resource.js (hunk 1 above)
Comment (66eli77) [inline kolibri/core/assets/src/api-resource.js:184]: at the beginning of this method, it checks `if (!this.id)` and reject a message if this.id is undefined.

Comment (rtibbles) [inline kolibri/core/assets/src/api-resource.js:184]: Oh, sorry - the `else` got hidden in the fold...

Comment (rtibbles) [inline kolibri/core/assets/src/api-resource.js:184]: But, you should probably change the reject message, as we can't pass in a model id now, yes?

Comment (66eli77) [inline kolibri/core/assets/src/api-resource.js:184]: yes, changed the reject message in the last commit.

**Comment by rtibbles** [comment]
This should do for now, methinks.

**Pull request merged**